src/js/health-records/containers/UpdatePage.jsx

```diff
@@ -0,0 +1,83 @@
+import React from 'react';
+import { connect } from 'react-redux';
+import { Link } from 'react-router';
+import { checkRefreshStatus } from '../actions/refresh';
+import { submitForm } from '../actions/form';
+
+import ProgressBar from '../../common/components/ProgressBar';
+
+export class UpdatePage extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.pollRefresh = () => {
+      setInterval(() => {
+        props.checkRefreshStatus();
+      }, 10000);
+    };
+    this.handleSkipToDownload = this.handleSkipToDownload.bind(this);
+  }
+
+  componentDidMount() {
+    this.props.checkRefreshStatus();
+    this.pollRefresh();
+  }
+
+  componentWillReceiveProps(nextProps) {
+    const erroredUpdates = nextProps.refresh.statuses.ERROR;
+    if (erroredUpdates.length === 0) {
+      this.props.submitForm(JSON.parse(sessionStorage.getItem('hr-form')));
+    }
+    if (nextProps.form.ready) {
+      this.context.router.push('/download');
+    }
+  }
+
+  componentWillUnmount() {
+    clearInterval(this.pollRefresh);
+  }
+
+  handleSkipToDownload(e) {
+    e.preventDefault();
+    this.props.submitForm(JSON.parse(sessionStorage.getItem('hr-form')));
+    this.context.router.push('/download');
+  }
+
+  render() {
+    const statuses = this.props.refresh.statuses;
+    const completionPercentage = statuses.OK.length / (statuses.ERROR.length + statuses.OK.length) * 100;
```

Comment:
Are these the proper metrics for the progress bar? Would it be more accurate to track the percentage of data with the last updated date being after the submission?